### PR TITLE
fix: single image with tag is handled incorrectly

### PR DESCRIPTION
**Pull request by thesayyn**

### CODE REVIEW

File: oci/tests/BUILD.bazel [hunk 1]
```diff
@@ -35,36 +31,17 @@ platform(
...
-    target_platform = _PLATFORM,
-)
-
-diff_test(
-    name = "test_java17",
-    file1 = "java17",
-    file2 = "distroless_java_amd64",
-)
-
-platform_transition_filegroup(
```
Comment (alexeagle) [inline oci/tests/BUILD.bazel:57]: how come we needed the transition before and don't need it now?

File: oci/private/util.bzl [hunk 2]
```diff
@@ -65,5 +65,6 @@ def _warning(rctx, message):
 
 util = struct(
     parse_image = _parse_image,
+    sha256 = _sha256,
```
Comment (alexeagle) [inline oci/private/util.bzl:68]: yay consistency

**Review by alexeagle — APPROVED** [review]

File: oci/tests/BUILD.bazel (hunk 1 above)
Comment (thesayyn) [reply oci/tests/BUILD.bazel:57]: this was added to get past the select statement we have in oci_alias and get the concrete oci_pull which would `amd64` that is equivalent to `{name}_linux_amd64`.

**Comment by thesayyn** [comment]
> Is there an issue fixed by it?

no. discovered it while going through the oci_pull code.

**Pull request merged**